### PR TITLE
support top-k search in test_performance

### PR DESCRIPTION
**Pull request by inabao**

### CODE REVIEW

File: tests/performance/README.md [hunk 1]
```diff
@@ -31,7 +32,7 @@ example running commands for searching:
 ```
 ./build/tests/test_performance \
     '/data/random-100k-128-euclidean.hdf5' \
-    'search' \
+    'search:1' \
```
Comment (wxyucs) [inline tests/performance/README.md:35]: @inabao add a new value k(or some other name) in the result json

Comment (inabao) [reply tests/performance/README.md:35]: done

**Review by wxyucs — APPROVED** [review]
lgtm

File: tests/performance/README.md [hunk 2]
```diff
@@ -2,7 +2,8 @@
 
 usage:
 ```
-Usage: ./build/tests/test_performance <dataset_file_path> <process> <index_name> <build_param> <search_param>
+build: ./build/tests/test_performance <dataset_file_path> "build" <index_name> <build_param> <search_param>
+search: ./build/tests/test_performance <dataset_file_path> "search:<top-k>" <index_name> <build_param> <search_param>
```
Comment (jiaweizone) [inline tests/performance/README.md:6]: Q: Why in search mode need provide `build_param` ？

File: tests/performance/test_performance.cpp [hunk 3]
```diff
@@ -393,16 +410,36 @@ class Test {
     }
 };
 
+bool
+valid_and_extrcat_top_k(const std::string& input, int64_t& number) {
+    std::regex pattern(R"(^search:(\d+)$)");
+    std::smatch match;
+    if (std::regex_match(input, match, pattern)) {
+        number = std::stoi(match[1].str());
+        if (number <= 0 || number > 100) {
```
Comment (jiaweizone) [inline tests/performance/test_performance.cpp]: Is 100 is too small?

File: tests/performance/test_performance.cpp [hunk 4]
```diff
@@ -393,16 +410,36 @@ class Test {
...
          const std::string& process,
          const std::string& index_name,
          const std::string& build_parameters,
          const std::string& search_parameters) {
+    int64_t top_k;
     if (process == "build") {
         return Test::Build(dataset_path, index_name, build_parameters);
-    } else if (process == "search") {
-        return Test::Search(dataset_path, index_name, build_parameters, search_parameters);
+    } else if (valid_and_extrcat_top_k(process, top_k)) {
```
Comment (jiaweizone) [inline tests/performance/test_performance.cpp:441]: Why not check `process == "search" `?

File: tests/performance/README.md (hunk 2 above)
Comment (inabao) [reply tests/performance/README.md:6]: Currently, the index does not support self-description, so the constructor of the index needs to be used before the serialization phase.

File: tests/performance/test_performance.cpp (hunk 4 above)
Comment (inabao) [reply tests/performance/test_performance.cpp:441]: We check the condition of `process == "search"` in `valid_and_extrcat_top_k(process, top_k)`.

Comment (inabao) [reply tests/performance/test_performance.cpp:441]: We check the condition of `process == "search"` in `valid_and_extrcat_top_k(process, top_k)`.

File: tests/performance/test_performance.cpp (hunk 3 above)
Comment (inabao) [reply tests/performance/test_performance.cpp]: Generally, when evaluating index performance, more attention is paid to the case where k is 100; however, this check condition can be removed.

**Review by jiaweizone — APPROVED** [review]
LGTM

**Pull request merged**